ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/group_vars/fabric_downlink_pools_missing.yml

```diff
@@ -0,0 +1,34 @@
+---
+fabric_name: fabric_downlink_pools_missing
+
+default_node_types:
+  - node_type: spine
+    match_hostnames:
+      - downlink-pools-missing-spine
+  - node_type: l3leaf
+    match_hostnames:
+      - downlink-pools-missing-l3leaf
+
+spine:
+  defaults:
+    loopback_ipv4_pool: 192.168.0.0/24
+    bgp_as: 65000
+  nodes:
+    - name: downlink-pools-missing-spine
+      id: 10
+
+l3leaf:
+  defaults:
+    loopback_ipv4_pool: 192.168.1.0/24
+    vtep_loopback_ipv4_pool: 192.168.2.0/24
+    bgp_as: 65000-65100
+    uplink_switches: [downlink-pools-missing-spine]
+    uplink_interfaces: [Ethernet1]
+  nodes:
+    - name: downlink-pools-missing-l3leaf
+      id: 10
+      uplink_switch_interfaces: [Ethernet3]
+
+expected_error_message: >-
+  Unable to assign IPs for uplinks. Either 'uplink_ipv4_pool' on this switch or 'downlink_pools' on
+  all the uplink switches is required but was not found for host 'downlink-pools-missing-l3leaf'
```

Comment:
I am also getting this error message if there are missing downlink interfaces for a node but the actual `uplink_ipv4_pool` or `downlink_pools` keys are present for this node:

```yaml
  node_groups:
    - group: SPINES
      nodes:
        - name: DC1-SPINE1
          id: 1
          mgmt_ip: 172.20.20.101/24
          downlink_pools:
            - downlink_ipv4_pool: 172.21.1.0/24
              downlink_interfaces: [Ethernet1-3]
        - name: DC1-SPINE2
          id: 2
          mgmt_ip: 172.20.20.102/24
          downlink_pools:
            - downlink_ipv4_pool: 172.21.2.0/24
              downlink_interfaces: [Ethernet1-3]
```
Here I have `[Ethernet1-3]` configured as downlinks but missing `Ethernet4` for an l3leaf with an ID of 4. I would expect to see a different error message if possible.

Reply:
Thanks for pointing it out, I raised a new error message when this scenario happens in [80d09f4](https://github.com/aristanetworks/avd/pull/3693/commits/80d09f4cad7bece07c78017c127d002a551eae47)